core/src/main/java/feast/core/config/SecurityConfig.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.config;
+
+import feast.core.CoreServiceGrpc;
+import feast.core.auth.authentication.GoogleOID.GoogleOpenIDAuthenticationProvider;
+import feast.core.auth.authorization.AuthorizationProvider;
+import feast.core.auth.authorization.Keto.KetoAuthorizationProvider;
+import feast.core.config.FeastProperties.SecurityProperties;
+import java.util.ArrayList;
+import java.util.List;
+import net.devh.boot.grpc.server.security.authentication.BearerAuthenticationReader;
+import net.devh.boot.grpc.server.security.authentication.GrpcAuthenticationReader;
+import net.devh.boot.grpc.server.security.check.AccessPredicate;
+import net.devh.boot.grpc.server.security.check.AccessPredicateVoter;
+import net.devh.boot.grpc.server.security.check.GrpcSecurityMetadataSource;
+import net.devh.boot.grpc.server.security.check.ManualGrpcSecurityMetadataSource;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.AccessDecisionManager;
+import org.springframework.security.access.AccessDecisionVoter;
+import org.springframework.security.access.vote.UnanimousBased;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.ProviderManager;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+
+@Configuration
+public class SecurityConfig {
+
+  private final SecurityProperties securityProperties;
+
+  public SecurityConfig(FeastProperties feastProperties) {
+    this.securityProperties = feastProperties.getSecurity();
+  }
+
+  /**
+   * Initializes an AuthenticationManager if authentication has been enabled.
+   *
+   * @return AuthenticationManager
+   */
+  @Bean
+  @ConditionalOnProperty(prefix = "feast.security.authentication", name = "enabled")
+  AuthenticationManager authenticationManager() {
+    final List<AuthenticationProvider> providers = new ArrayList<>();
+
+    if (securityProperties.getAuthentication().isEnabled()) {
+      switch (securityProperties.getAuthentication().getProvider()) {
+        case "GoogleOpenID":
+          providers.add(
+              new GoogleOpenIDAuthenticationProvider(
```

Comment:
Why not use the capabilities of spring to configure the providers rather than build another factory method?

Reply:
> Does this really only support Google OpenID? It shouldn't be any effort to make the OIDC configurable to support any valid implementation such as Auth0.

Yip, it should be pretty easy to extend to non-Google through config. This was a minimal implementation. The reason I specified it as Google is because our Python SDK uses OIDC tokens that it gets from a local Gcloud SDK and sends that along to Feast for auth. That is very Google specific.

Server side can easily be generalized.

Reply:
Can be done. See points above.